org-brain.el

```diff
@@ -62,24 +64,94 @@ This will be used by `org-brain-new-child'."
   :group 'org-brain
   :type '(string))
 
+;;; Utils
+(defun flatten (obj)
+  (do* ((result (list obj))
+        (node result))
+       ((null node) (delete nil result))
+    (cond ((consp (car node))
+           (when (cdar node) (push (cdar node) (cdr node)))
+           (setf (car node) (caar node)))
+          (t (setf node (cdr node))))))
+
+;;; Logging
+(defcustom org-brain-log t
+  "Set to nil to not write to *Messages* buffer."
+  :group 'org-brain
+  :type 'boolean)
+
+(defun org-brain-log (msg)
+  (when org-brain-log
+    (setq inhibit-message t)
+    (message msg)
+    (setq inhibit-message nil)))
+
+;;; Caches
```

Comment:
Yeah, defvar is tricky. In Common Lisp it doesn't allow change once set, I think it works the same in elisp. defvar would allow docstrings, but simple variables don't, at least I don't think they do. In Common Lisp we'd have used defparameter for this type of variable, but it seems elisp has no equivalent.